ddtrace/tracer/textmap_test.go

```diff
@@ -355,210 +355,244 @@ func TestTextMapPropagator(t *testing.T) {
 	})
 }
 
-func testB3(t *testing.T, b3Header string) {
-	t.Run("inject", func(t *testing.T) {
-		os.Setenv("DD_PROPAGATION_STYLE_INJECT", b3Header)
-		defer os.Unsetenv("DD_PROPAGATION_STYLE_INJECT")
-
-		var tests = []struct {
-			in  []uint64
-			out map[string]string
-		}{
-			{
-				[]uint64{1412508178991881, 1842642739201064},
-				map[string]string{
-					b3TraceIDHeader: "000504ab30404b09",
-					b3SpanIDHeader:  "00068bdfb1eb0428",
+func TestEnvVars(t *testing.T) {
+	var testEnvs []map[string]string
+
+	testEnvs = []map[string]string{
+		{headerPropagationStyleInject: "b3"},
+		{headerPropagationStyleInjectDeprecated: "b3,none" /* none should have no affect */},
+		{headerPropagationStyle: "b3"},
+		{headerPropagationStyleInject: "b3multi", headerPropagationStyleInjectDeprecated: "none" /* none should have no affect */},
+		{headerPropagationStyleInject: "b3multi", headerPropagationStyle: "none" /* none should have no affect */},
+	}
+	for _, testEnv := range testEnvs {
+		t.Run(fmt.Sprintf("inject with env=%q", testEnv), func(t *testing.T) {
+			for k, v := range testEnv {
+				os.Setenv(k, v)
```

Comment:
Small nit about os.Setenv (can be done in a separate PR if applicable). Can use `t.Setenv(k,v)`

```
Setenv calls os.Setenv(key, value) and uses Cleanup to restore the environment variable to its original value after the test.
This cannot be used in parallel tests.```